pkg/checks/startup.go

```diff
@@ -0,0 +1,65 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dbtuneai/agent/pkg/agent"
+	"github.com/dbtuneai/agent/pkg/dbtune"
+	"github.com/dbtuneai/agent/pkg/pg"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+// CheckStartupRequirements verifies all mandatory configuration and connectivity requirements.
+// Returns nil if all checks pass, or an error describing the first failure.
+func CheckStartupRequirements() error {
+	// Check PostgreSQL connection URL
+	pgConfig, err := pg.ConfigFromViper(nil)
+	if err != nil {
+		return fmt.Errorf("PostgreSQL connection URL not set or invalid: %w", err)
+	}
+	if pgConfig.ConnectionURL == "" {
+		return fmt.Errorf("PostgreSQL connection URL is required but not set")
+	}
+
+	// Check dbtune server URL, API key, and database ID
+	serverURLs, err := dbtune.CreateServerURLs()
+	if err != nil {
+		return fmt.Errorf("dbtune server configuration invalid: %w", err)
+	}
+	if serverURLs.ServerUrl == "" {
+		return fmt.Errorf("dbtune server URL is required but not set")
+	}
+	if serverURLs.ApiKey == "" {
+		return fmt.Errorf("dbtune API key is required but not set")
+	}
+	if serverURLs.DbID == "" {
+		return fmt.Errorf("dbtune database ID is required but not set")
+	}
```

Comment:
Can we match these messages to the actual field names required?
https://github.com/dbtuneai/dbtune-agent/blob/ea58b899e15065b848fef3341cd6ffed89a67a8f/pkg/dbtune/urls.go#L14-L18

Reply:
Mmmm I think it's a bit more subtle, in that it only detects that it's present, not necessarily for blank strings. However given we use the actual values for checking, I think the removal is good.